sources/common-dylan/byte-vector.dylan

```diff
@@ -221,3 +221,42 @@ define sealed inline method byte-storage-offset-address
            (primitive-repeated-slot-as-raw
               (the-buffer, primitive-repeated-slot-offset(the-buffer)))))
 end method;
+
+define method from-hexstring (string :: <byte-string>)
+  => (result :: <byte-vector>)
+  if (odd?(string.size))
+    error("String size must be multiple of 2.");
+  end if;
+
+  let result :: <byte-vector> = make(<byte-vector>, size: floor/(string.size, 2));
+
+  without-bounds-checks
+    for (i :: <integer> from 0 below string.size - 1 by 2)
+      let hi :: <byte> = as(<byte>, string[i]);
+      let lo :: <byte> = as(<byte>, string[i + 1]);
+      let b  :: <byte> = ash(logand(hi, 15) + 9 * ash(hi, -6), 4) +
+        logand(lo, 15) + 9 * ash(lo, -6);
```

Comment:
My indenting is done using dylan-mode for emacs. I'm fine with correcting this, but is there a way I can instruct emacs to follow this general rule?